fxa-auth-db-server/test/backend/db_tests.js

```diff
@@ -2160,6 +2161,35 @@ module.exports = function(config, DB) {
               )
               .then(
                 function() {
+                  // Can create an account from a deleted account's secondary email
+                  const testAccount = createAccount()
+                  testAccount.emailVerified = true
+
+                  const testSecondaryEmail = createEmail({
+                    uid: testAccount.uid,
+                    isVerified: true
+                  })
+
+                  return db.createAccount(testAccount.uid, testAccount)
+                    .then(function () {
+                      return db.createEmail(testAccount.uid, testSecondaryEmail)
+                    })
+                    .then(function () {
+                      return db.deleteAccount(testAccount.uid)
+                    })
+                    .then(function () {
+                      testAccount.email = testSecondaryEmail.email
+                      testAccount.normalizedEmail = testSecondaryEmail.normalizedEmail
+                      return db.createAccount(testAccount.uid, testAccount)
```

Comment:
after this createAccount we also need `db.createEmail(` again to see if the `testSecondaryEmail` can be used (and if it was really deleted)